checkbox-provider-ce-oem/bin/thermal_sensor_test.py

```diff
@@ -0,0 +1,177 @@
+    date_format = "%Y-%m-%d %H:%M:%S"
+
+    # Log DEBUG and INFO to stdout, others to stderr
+    stdout_handler = logging.StreamHandler(sys.stdout)
+    stdout_handler.setFormatter(logging.Formatter(logger_format, date_format))
+
+    stderr_handler = logging.StreamHandler(sys.stderr)
+    stderr_handler.setFormatter(logging.Formatter(logger_format, date_format))
+
+    stdout_handler.setLevel(logging.DEBUG)
+    stderr_handler.setLevel(logging.WARNING)
+
+    # Add a filter to the stdout handler to limit log records to
+    # INFO level and below
+    stdout_handler.addFilter(lambda record: record.levelno <= logging.INFO)
+
+    root_logger.addHandler(stderr_handler)
+    root_logger.addHandler(stdout_handler)
+
+    return root_logger
+
+
+class ThermalMonitor():
+
+    def __init__(self, name):
+        self._name = name
+        self.root_node = Path(SYS_THERMAL_PATH).joinpath(self._name)
+        self.type_node = self.root_node.joinpath("type")
+        self.temp_node = self.root_node.joinpath("temp")
+        self.mode_node = self.root_node.joinpath("mode")
+        self.initial_temp = None
+
+    def _read_node(self, node):
+        if node.exists():
+            return node.read_text().strip("\n")
+        else:
+            raise FileNotFoundError("{} file not exists".format(str(node)))
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def type(self):
+        return self._read_node(self.type_node)
+
+    @property
+    def temperature(self):
+        temp = self._read_node(self.temp_node)
+        if not temp.isnumeric():
+            raise ValueError("temperate value is not a number!")
+        return temp
+
+    @property
+    def mode(self):
+        return self._read_node(self.mode_node)
+
+
+def check_temperature(current, initial):
+    logging.info("Initial value: %s, current value: %s", initial, current)
+    return (int(current) != 0 and current != initial)
+
+
+def thermal_monitor_test(args):
+    logging.info(
+        "# Monitor the temperature of %s thermal around %s seconds",
+        args.name, args.duration)
+
+    if args.extra_commands == "stress-ng":
+        cmd = "stress-ng --cpu 0 --io 4 --vm 2 --vm-bytes 128M --timeout 60s"
+    else:
+        cmd = args.extra_commands
+
+    result = False
+    try:
+        thermal_op = ThermalMonitor(args.name)
+        initial_value = thermal_op.temperature
+
+        with subprocess.Popen(shlex.split(cmd)) as proc:
+            for _ in range(args.duration):
+                cur_temp = thermal_op.temperature
+                result = check_temperature(cur_temp, initial_value)
+                if result:
+                    logging.info(
+                        "# The temperature of %s thermal has been altered",
+                        args.name)
+                    break
+                time.sleep(1)
+            proc.kill()
+    except subprocess.CalledProcessError:
+        # Bypass the subprocess error
```

Comment:
May I know why we don't do a thing when error detected?

Reply:
that's because of we are trying to increase the system loading by subprocess, and the major purpose is to make sure the thermal temperature works properly, so we could safety ignore any error raises by subprocess.

I also added the notes about why we can bypass the subprocess error.